lms/djangoapps/courseware/views/views.py

```diff
@@ -1970,6 +1962,48 @@ def financial_assistance_request(request):
     return HttpResponse(status=status.HTTP_204_NO_CONTENT)
 
 
+@login_required
+@require_POST
+def financial_assistance_request_v2(request):
+    """
+    Uses the new financial assistance application flow.
+    Creates a post request to edx-financial-assistance backend.
+    """
+    try:
+        data = json.loads(request.body.decode('utf8'))
+        username = data['username']
+        # Simple sanity check that the session belongs to the user
+        # submitting an FA request
+        if request.user.username != username:
+            return HttpResponseForbidden()
+
+        lms_user_id = request.user.id
+        course_id = data['course']
+        income = data['income']
+        learner_reasons = data['reason_for_applying']
+        learner_goals = data['goals']
+        learner_plans = data['effort']
+        allowed_for_marketing = data['mktg-permission']
+
+    except ValueError:
+        # Thrown if JSON parsing fails
+        return HttpResponseBadRequest('Could not parse request JSON.')
+    except KeyError as err:
```

Comment:
do we need a general exception as well?

Reply:
I don't think so. All the cases have been handled.